autorest/azure/rp.go

```diff
@@ -0,0 +1,195 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// RegisterResourceProvider  tries to register the resource provider in case it is unregistered
+func RegisterResourceProvider(client autorest.Client) autorest.SendDecorator {
+	return func(s autorest.Sender) autorest.Sender {
+		return autorest.SenderFunc(func(r *http.Request) (resp *http.Response, err error) {
+			rr := autorest.NewRetriableRequest(r)
+			err = rr.Prepare()
+			if err != nil {
+				return resp, err
+			}
+
+			resp, err = autorest.SendWithSender(s, rr.Request(),
+				autorest.DoRetryForStatusCodes(client.RetryAttempts, client.RetryDuration, autorest.StatusCodesForRetry...),
+			)
+			if err != nil {
+				return resp, err
+			}
+
+			if resp.StatusCode == http.StatusConflict {
+				var re RequestError
+				err = autorest.Respond(
+					resp,
+					autorest.ByUnmarshallingJSON(&re),
+				)
+				if err != nil {
+					return resp, err
+				}
+
+				if re.ServiceError != nil && re.ServiceError.Code == "MissingSubscriptionRegistration" {
+					// in tests, the sender should be the same for all requests
+					client.Sender = s
+					err = register(client, r, re)
+					if err != nil {
+						return resp, fmt.Errorf("failed auto registering Resource Provider: %s", err)
+					}
+				}
+				err = rr.Prepare()
```

Comment:
I think this code and the lines below it can be removed, refactoring the code starting at line 19 into a loop.

Reply:
Fixed